support/logging-client/logger.go

```diff
@@ -32,7 +32,15 @@ import (
 	"strings"
 )
 
-type LoggingClient struct {
+type LoggingClient  interface {
+	Debug(msg string, labels ...string) error
```

Comment:
Why this definition in the interface? Why not copying the real functions signature (logLevel string, msg string, labels []string)?

Reply:
Because the method names ARE the logLevel. The interface provides a discrete contract for what those log levels are so that calling logic doesn't pass in an unsupported value.

Anyway, calling logic throughout the application doesn't call the LoggingClient as
loggingClient.log("Info",fmt.Sprintf("Starting %s %s ", command.COMMANDSERVICENAME, edgex.Version))

"log" is an internal method on the EdgeXLogger implementation class. Everywhere the use of LoggingClient is done via:
loggingClient.Info(fmt.Sprintf("Starting %s %s ", command.COMMANDSERVICENAME, edgex.Version))

Thus the interface is defined accordingly.

Reply:
Sorry,  you are right. I misunderstand the code because of the 'labels ...string'.  Still too much go to learn.